cli/src/main/java/org/openjdk/skara/cli/GitWebrev.java

```diff
@@ -158,7 +163,7 @@ private static void generate(String[] args) throws IOException {
         }
 
         var cwd = Paths.get("").toAbsolutePath();
-        var repository = ReadOnlyRepository.get(cwd);
+        var repository = Repository.get(cwd);
```

Comment:
What non-read-only method did you need? Can't spot it. :)

Reply:
To figure out the outgoing commits I need to find the closest merge-base with regards to all upstream branches. Since it is not certain that all upstream branches exists locally, I need to fetch the upstream branch to `FETCH_HEAD`. This is not a destructive operation for the local repository, it only mutates `FETCH_HEAD` which you shouldn't rely on anyhow staying the same.

The `fetch` method is only available on `Repository`, so I had to use `Repository` instead of `ReadOnlyRepository`.